### PR TITLE
libs: protection: x86_64: drop root requirement for querying

### PR DESCRIPTION
**Pull request by jodh-intel**
It is no longer necessary to be `root` to query the guest protection
(TDX) on `x86_64` systems, so drop the requirement.

Also update `kata-ctl` to remove the duplicate, and now redundant, root check.

Fixes: #8548.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: src/libs/kata-sys-util/src/protection.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use std::path::Path;
 use std::path::PathBuf;
 use thiserror::Error;
 
-#[cfg(any(target_arch = "s390x", target_arch = "x86_64"))]
+#[cfg(any(target_arch = "s390x", target_arch = "powerpc64le"))]
```
Comment (fidencio) [inline src/libs/kata-sys-util/src/protection.rs:16]: Out of curiosity, why powerpc64le was added here?

It seems not related to this commit.

Comment (jodh-intel) [reply src/libs/kata-sys-util/src/protection.rs:16]: I've added a comment to the commit. Basically PPC64 needs this import fwics but it was previously missing.

**Review by amshinde — APPROVED** [review]

**Review by cmaf — APPROVED** [review]
@jodh-intel LGTM.
